Modern/ops/src/main/java/org/bonej/ops/ellipsoid/HalfOrMoreSamplingPointsOutOfBounds.java

```diff
@@ -0,0 +1,59 @@
+package org.bonej.ops.ellipsoid;
+
+
+import net.imagej.ops.Op;
+import net.imagej.ops.special.function.AbstractBinaryFunctionOp;
+import net.imglib2.Dimensions;
+import net.imglib2.img.Img;
+import org.bonej.utilities.VectorUtil;
+import org.joml.Matrix3d;
+import org.joml.Vector3d;
+import org.joml.Vector3dc;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+
+import java.util.Arrays;
+import java.util.List;
+
+@Plugin(type = Op.class)
+public class HalfOrMoreSamplingPointsOutOfBounds extends AbstractBinaryFunctionOp<Img,Ellipsoid,Boolean> {
+
+    @Parameter(required = false)
+    private List<Vector3dc> samplingDirections = Arrays.asList(
+            new Vector3d(1, 0, 0), new Vector3d(0, 1, 0), new Vector3d(0, 0, 1),
+                new Vector3d(-1, 0, 0), new Vector3d(0, -1, 0), new Vector3d(0, 0, -1));;
+
+    @Override
+    public Boolean calculate(Img img, Ellipsoid ellipsoid) {
+        final Matrix3d a = new Matrix3d();
+        ellipsoid.getOrientation().get3x3(a);
+        final Vector3dc centroid = ellipsoid.getCentroid();
+        final long surfacePointsOutside = samplingDirections.stream().filter(dir -> {
```

Comment:
This seems a bit heavyweight - surely you have the surface points already somewhere, and it looks like you are calculating them all again. If it works for now, keep it, but it would be more efficient that you kill an ellipsoid in the same loop where you check whether its surface points are inside or outside - just keep a count of inside vs outside in that loop and if at the end outside > inside set that ellipsoid to null or otherwise remove it from the calculations, immediately. That way you only have to calculate the ellipsoid and check each point once.